conformance/lang/expressions/range-expr/range_expr_exclusive.balt

```diff
@@ -0,0 +1,5877 @@
+}
+
+Test-Case: output
+Description: Test isolatedness of the iterator() and next() methods of range expression (exclusive).
+Fail-Issue: ballerina-platform/ballerina-lang#33163
+Labels: range-expr, object-type, array-type, list-constructor-expr, array:push, int, DecimalNumber, record-type,
+        optional-type, is-expr, method-call-expr
+
+function init() {
+    object {
+        public isolated function iterator() returns (object {
+            public isolated function next() returns (record {|int value;|}?);
+        });
+    } range = 1 ..< 10;
+
+
+    io:println(<any>range.iterator() is isolated function); // @output true
+    io:println(<any>range.iterator().next() is isolated function); // @output true
+}
+
+Test-Case: output
+Description: Test range expression (exclusive) with query expression.
+Labels: range-expr, array-type, list-constructor-expr, array:push, int, DecimalNumber, query-expr
+
+function init() {
+    int[] a = from int i in 1 ..< 10
+              select i;
+
+    io:println(a); // @output [1,2,3,4,5,6,7,8,9]
+}
+
+Test-Case: error
+Description: Test range expression (exclusive) when static type of either of the expression is not a subtype of int.
+Labels: range-expr, int, DecimalFloatingPointNumber, string
+
+function init() {
+    _ = 1.0  ..<  10; // @error  ..<  not defined when static type of either of the expression is not a subtype of int
+    _ = 1  ..<  10.0; // @error  ..<  not defined when static type of either of the expression is not a subtype of int
+    _ = 1.0  ..<  10.0; // @error  ..<  not defined when static type of either of the expression is not a subtype of int
+
+    _ = "1"  ..<  10; // @error  ..<  not defined when static type of either of the expression is not a subtype of int
+    _ = 1  ..<  "10"; // @error  ..<  not defined when static type of either of the expression is not a subtype of int
+    _ = "1"  ..<  "10"; // @error  ..<  not defined when static type of either of the expression is not a subtype of int
+}
+
+Test-Case: error
+Description: Test assigning the result of range expression (exclusive) to an incompatible type.
+Labels: range-expr, int, DecimalNumber, object-type, float, record-type, optional-type
+
+function init() {
+    int _ = 1  ..<  10; // @error result of a range expression is an object
+
+    object {
+        public isolated function iterator() returns (object {
+            public isolated function next() returns (record {|float value;|}?);
+        });
+    } _ = 1  ..<  10; // @error next() of range expression's iterable object returns a record {|int value;|}
+}
+
+Test-Case: error
+Description: Test assigning the result of range expression (exclusive) to an incompatible type when the static
+             type of the range expression is inferred.
+Labels: range-expr, int, DecimalNumber, object-type, var, record-type, optional-type, float
+
+function init() {
+    var a = 1  ..<  10;
+
+    int _ = a; // @error result of a range expression is an object
```

Comment:
Shall we include the bit about expecting an `int` here. Basically we should be able to get some idea about whether the actual error is what we expected by comparing just this and the actual error message (i.e., without looking at the code).